File: front-end/src/app/service-models/app/v1/client/client-employee-pay-slips-s-m.ts
import { ClientCompanyDetailSM } from './client-company-detail-s-m';
import { ClientEmployeePayrollComponentSM } from './client-employee-payroll-component-s-m';

export class ClientEmployeePaySlipsSM extends ClientCompanyDetailSM {
    employeeName!: string;
    dateOfJoining!: Date;
    employeeEmail!: string;
    employeePhone!: string;
    designation!: string;
    paymentFor!: string;
    paymentAmount!: number;
    paymentPaid!: boolean;
    employeeCode!: string;
    ctcAmount!: number;
    companyAddress!: string;
    clientEmployeePayrollComponents!: Array<ClientEmployeePayrollComponentSM>;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/Client/ClientEmployeePaySlipsSM.cs
using System.Collections.Generic;

namespace Siffrum.Web.Payroll.ServiceModels.v1.Client
{
    public class ClientEmployeePaySlipsSM : ClientCompanyDetailSM
    {
        public string EmployeeName { get; set; }
        public DateTime DateOfJoining { get; set; }
        public string EmployeeEmail { get; set; }
        public string EmployeePhone { get; set; }
        public string Designation { get; set; }
        public string PaymentFor { get; set; }
        public float PaymentAmount { get; set; }
        public bool PaymentPaid { get; set; }
        public string EmployeeCode { get; set; }
        public float CtcAmount { get; set; }
        public string CompanyAddress { get; set; }
        public List<ClientEmployeePayrollComponentSM> ClientEmployeePayrollComponents { get; set; }

    }
}
